### PR TITLE
spadd perf test, nonsquare MatrixMarket output

### PR DESCRIPTION
**Pull request by brian-kelley**
- Add a performance test for spadd, that operates on user-provided matrices. Initially created to try to replicate a bug observed in Trilinos, but might as well keep it. Would be useful for measuring benchmarks for the wiki.
- IOUtils: Make write_kokkos_crst_matrix support non-square matrices, for matrix-market output. Before, it just used mat.numRows() and silently assumed it's square. This meant invalid memory accesses if numCols < numRows, and an incomplete dump if numCols > numRows. The bin, ligra and crs formats still only support square but now an exception is thrown if the matrix is non-square.

TODO: update the other 3 matrix formats too

White spot check:
#######################################################
PASSED TESTS
#######################################################
cuda-10.0.130-Cuda_Serial-release build_time=1129 run_time=1492
cuda-9.2.88-Cuda_OpenMP-release build_time=1214 run_time=1109
gcc-6.4.0-OpenMP_Serial-release build_time=597 run_time=1262
gcc-7.2.0-OpenMP-release build_time=405 run_time=482
gcc-7.2.0-OpenMP_Serial-release build_time=624 run_time=1504
gcc-7.2.0-Serial-release build_time=272 run_time=647
gcc-7.4.0-OpenMP-release build_time=446 run_time=516
ibm-16.1.0-Serial-release build_time=1316 run_time=1003
#######################################################
FAILED TESTS
#######################################################

Bowman spot-check:
#######################################################
PASSED TESTS
#######################################################
intel-16.4.258-Pthread-release build_time=1634 run_time=2991
intel-16.4.258-Pthread_Serial-release build_time=2765 run_time=5865
intel-16.4.258-Serial-release build_time=1589 run_time=2784
intel-17.2.174-OpenMP-release build_time=2108 run_time=1086
intel-17.2.174-OpenMP_Serial-release build_time=2952 run_time=4314
intel-17.2.174-Pthread-release build_time=1424 run_time=3035
intel-17.2.174-Pthread_Serial-release build_time=2568 run_time=5620
intel-17.2.174-Serial-release build_time=1473 run_time=2756
intel-18.2.199-OpenMP-release build_time=1693 run_time=1108
intel-18.2.199-OpenMP_Serial-release build_time=2982 run_time=4007
intel-18.2.199-Pthread-release build_time=1315 run_time=3120
intel-18.2.199-Pthread_Serial-release build_time=2604 run_time=5670
intel-18.2.199-Serial-release build_time=1292 run_time=2869
#######################################################
FAILED TESTS
#######################################################

### CODE REVIEW

File: perf_test/sparse/KokkosSparse_run_spadd.hpp [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  using namespace KokkosSparse::Experimental;
+
+  using size_type = typename crsMat_t::size_type;
+  using lno_t = typename crsMat_t::ordinal_type;
+  using scalar_t = typename crsMat_t::value_type;
+  using device_t = typename crsMat_t::device_type;
+  using exec_space = typename device_t::execution_space;
+  using mem_space = typename device_t::memory_space;
+
+  using KernelHandle = KokkosKernels::Experimental::KokkosKernelsHandle
```
Comment (mhoemmen) [inline perf_test/sparse/KokkosSparse_run_spadd.hpp]: Are these things still in the Experimental namespace? or have they been imported into KokkosKernels?

Comment (brian-kelley) [reply perf_test/sparse/KokkosSparse_run_spadd.hpp]: @mhoemmen Yes, all the sparse kernels are still in this namespace.

Comment (mhoemmen) [reply perf_test/sparse/KokkosSparse_run_spadd.hpp]: @brian-kelley So is @trilinos/muelu depending nonoptionally on experimental code?

Comment (brian-kelley) [reply perf_test/sparse/KokkosSparse_run_spadd.hpp]: @mhoemmen Technically yes, and Tpetra as well (spgemm/spadd). But really, these things are all several years old and have stable interfaces, so we should talk about removing the Experimental namespace.

**Pull request merged**